apiml-utility/src/test/java/org/zowe/apiml/product/logging/LogMessageTrackerTest.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+package org.zowe.apiml.product.logging;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.regex.Pattern;
+
+import static junit.framework.TestCase.assertFalse;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+class LogMessageTrackerTest {
+    private static final String LOG_MESSAGE = "This is a log message.";
+    private static final String NOT_LOGGED_MESSAGE = "This is not a log message.";
+
+    private static final Pattern MESSAGE_REGEX = Pattern.compile("^This.*");
+    private static final Pattern NOT_MESSAGE_REGEX = Pattern.compile("^dummy");
+
+    private final LogMessageTracker logMessageTracker = new LogMessageTracker(this.getClass());
+    private final Logger log = (Logger) LoggerFactory.getLogger(this.getClass());
+
+    @BeforeEach
+    void setup() {
+        logMessageTracker.startTracking();
+        log.trace(LOG_MESSAGE);
+        log.debug(LOG_MESSAGE);
+        log.info(LOG_MESSAGE);
+        log.warn(LOG_MESSAGE);
+        log.error(LOG_MESSAGE);
+    }
+
+    @AfterEach
+    void cleanUp() {
+        logMessageTracker.stopTracking();
+    }
+
+    @Test
+    void testSearch() {
```

Comment:
We are trying to keep the given_when_then structure in the test names, would it be possible to have it here as well?

Reply:
Yes I will change them.